core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -513,3 +513,81 @@ class CollectionAccumulator[T] extends AccumulatorV2[T, java.util.List[T]] {
     getOrCreate.addAll(newValue)
   }
 }
+
+
+/**
+ * An [[AccumulatorV2 counter]] for collecting a list of (mapper id, row count).
+ *
+ * @since 3.4.0
+ */
+class MapperRowCounter extends AccumulatorV2[jl.Long, java.util.List[java.util.List[jl.Long]]] {
```

Comment:
We can put it in the sql module, or probably the same file with shuffle node.

Reply:
Moved to sql module.